src/SHA256.mo

```diff
@@ -134,17 +134,18 @@ module {
     };
 
     private func block(data : [Nat8]) {
-      var p = data;
       var w = Array.init<Nat32>(64, 0);
-      while (p.size() >= 64) {
+      var start = 0;
+      var len = data.size();
+      while (len >= 64) {
         var j = 0;
         for (i in Iter.range(0, 15)) {
```

Comment:
The for loop might be quicker as a while loop starting j from start, incrementing by 4 up to a pre-computed bound.

Reply:
But `i` has to increment, too.
Why exactly would it be quicker?